From 52c53b383c0f4b07ebbb53e0dd2fc6c5bb051cf9 Mon Sep 17 00:00:00 2001
From: Houjun Liu <hliu@shabang.cf>
Date: Mon, 13 Jun 2022 22:59:32 -0700
Subject: [PATCH] Change necessary to make HTK-3.4.1 work for P2FA

---
 HTKLib/HRec.c   | 4 ++--
 HTKLib/strarr.c | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/HTKLib/HRec.c b/HTKLib/HRec.c
index 8323130..a557dd5 100644
--- a/HTKLib/HRec.c
+++ b/HTKLib/HRec.c
@@ -1623,7 +1623,7 @@ static void LatFromPaths(Path *path,int *ln,Lattice *lat)
             la->lAlign[i].label=labid;
 #ifdef PHNALG
 	    /* didn't handle model that allow 0 frame */
-            if (dur<=0 && labid != splabid) HError(8522,"LatFromPaths: Align  have dur<=0");
+            /*if (dur<=0 && labid != splabid) HError(8522,"LatFromPaths: Align  have dur<=0");*/
 #endif
             la->lAlign[i].dur=dur;
             la->lAlign[i].like=like;
@@ -1647,7 +1647,7 @@ static void LatFromPaths(Path *path,int *ln,Lattice *lat)
             la->lAlign[i].label=labpr;
 #ifdef PHNALG
 	    /* didn't handle model that allow 0 frame */
-            if (dur<=0 && labid != splabid) HError(8522,"LatFromPaths: Align have dur<=0 ");
+            /*if (dur<=0 && labid != splabid) HError(8522,"LatFromPaths: Align have dur<=0 ");*/
 #endif
             la->lAlign[i].dur=dur;
             la->lAlign[i].like=like;
diff --git a/HTKLib/strarr.c b/HTKLib/strarr.c
index ea5a930..e9e44ae 100644
--- a/HTKLib/strarr.c
+++ b/HTKLib/strarr.c
@@ -18,7 +18,7 @@
 /* ----------------------------------------------------------- */
 
 
-#include <malloc.h>
+#include <malloc/malloc.h>
 #include "esignal.h"
 #include "strarr.h"
 /*
-- 
2.30.1 (Apple Git-130)

